server/api/users.js

```diff
@@ -12,3 +12,37 @@ router.get('/', (req, res, next) => {
     .then(users => res.json(users))
     .catch(next)
 })
+
+router.post('/', (req, res, next) => {
+  User.findOrCreate(req.body)
+    .then(user => res.status(201).json(user))
+    .catch(next)
+})
+
+router.get('/:userId', (req, res, next) => {
+  User.findOne({
+    attributes: ['id', 'email'],
+    where: {
+      id: req.params.userId
+    }
+  })
+    .then(user => res.json(user))
+    .catch(next)
+})
+
+router.put('/:userId', (req, res, next) => {
+  User.findById(req.params.userId)
+    .then(user => user.update(req.body))
+    .then(newUser => res.json(newUser.dataValues))
```

Comment:
do you need to get the dataValues attricute or just newUser?

Reply:
both works! but I'll take out dataValues just to stay consistent